src/main/webapp/cluster-view/cluster-view.controller.js

```diff
@@ -1,26 +1,150 @@
 'use strict';
 
 angular.module('managementConsole')
-    .controller('ClusterViewCtrl', [
+  .controller('ClusterViewCtrl', [
     '$scope',
     '$stateParams',
     '$state',
+    '$q',
     'modelController',
-    function ($scope, $stateParams, $state, modelController) {
-            if (!modelController.isAuthenticated()) {
-                $state.go('/logout');
+    function ($scope, $stateParams, $state, $q, modelController) {
+      if (!modelController.isAuthenticated()) {
```

Comment:
@vblagoje: we should not check if user is authenticated here because it is too late then, controller will already start rendering and there will be flashing and similar which is not nice. We should listen to state change instead ($stateChangeStart), and on each state change trigger logic that will check if user has permission to go that state or not.